src/dataflow/src/source/timestamp.rs

```diff
@@ -268,15 +268,23 @@ pub struct TimestampBindingBox {
     durability_frontier: Antichain<Timestamp>,
     /// Generates new timestamps for RT sources
     proposer: Option<TimestampProposer>,
+    /// Never persist these bindings. This is used for BYO, where the bindings
+    /// are stored externally already.
+    never_requires_persistence: bool,
```

Comment:
this boolean seems more useful for kinesis et al because byo we can handle better by setting the durability frontier to +inf. I need to think about this a bit more to see if maybe we can unify both approaches -- having two ways to disable ts binding persistence seems tricky

Reply:
You mean, set the durability frontier to +Inf and then check the durability frontier before sending bindings from the worker to the coordinator? Because right now the worker always sends the bindings.

Reply:
yeah exactly -- but this has the side effect then that downstream readers of the durability frontier could reasonably conclude that the stream is fully durable which is definitely not true for something like kinesis :/

Reply:
I think if you only want to disable persistence for BYO sources then yeah setting the durability frontier at creation time and then checking it and only sending durability updates if its in advance of the frontier is the way to go

Reply:
Hmm, I'd probably merge the current state, which is a bit complicated. With the understanding that timestamping and the durability frontier might change again soonishly. What do you think?